pcmanfm/desktopitemdelegate.cpp

```diff
@@ -32,172 +32,178 @@
 namespace PCManFM {
 
 DesktopItemDelegate::DesktopItemDelegate(QListView* view, QObject* parent):
-  QStyledItemDelegate(parent ? parent : view),
-  view_(view),
-  symlinkIcon_(QIcon::fromTheme("emblem-symbolic-link")),
-  shadowColor_(0, 0, 0),
-  margins_(QSize(3, 3)) {
+    QStyledItemDelegate(parent ? parent : view),
+    view_(view),
+    symlinkIcon_(QIcon::fromTheme("emblem-symbolic-link")),
+    shadowColor_(0, 0, 0),
+    margins_(QSize(3, 3)) {
 }
 
 // FIXME: we need to figure out a way to derive from Fm::FolderItemDelegate to avoid code duplication.
 void DesktopItemDelegate::paint(QPainter* painter, const QStyleOptionViewItem& option, const QModelIndex& index) const {
-  Q_ASSERT(index.isValid());
-  QStyleOptionViewItem opt = option;
-  initStyleOption(&opt, index);
-
-  painter->save();
-  painter->setClipRect(option.rect);
-
-  opt.decorationAlignment = Qt::AlignHCenter | Qt::AlignTop;
-  opt.displayAlignment = Qt::AlignTop | Qt::AlignHCenter;
-
-  // draw the icon
-  QIcon::Mode iconMode;
-  if(opt.state & QStyle::State_Enabled) {
-    if(opt.state & QStyle::State_Selected)
-      iconMode = QIcon::Selected;
+    Q_ASSERT(index.isValid());
+    QStyleOptionViewItem opt = option;
+    initStyleOption(&opt, index);
+
+    painter->save();
+    painter->setClipRect(option.rect);
+
+    opt.decorationAlignment = Qt::AlignHCenter | Qt::AlignTop;
+    opt.displayAlignment = Qt::AlignTop | Qt::AlignHCenter;
+
+    // draw the icon
+    QIcon::Mode iconMode;
+    if(opt.state & QStyle::State_Enabled) {
+        if(opt.state & QStyle::State_Selected) {
+            iconMode = QIcon::Selected;
+        }
+        else {
+            iconMode = QIcon::Normal;
+        }
+    }
     else {
-      iconMode = QIcon::Normal;
+        iconMode = QIcon::Disabled;
     }
-  }
-  else
-    iconMode = QIcon::Disabled;
-  QPoint iconPos(opt.rect.x() + (opt.rect.width() - opt.decorationSize.width()) / 2, opt.rect.y());
-  QPixmap pixmap = opt.icon.pixmap(opt.decorationSize, iconMode);
-  painter->drawPixmap(iconPos, pixmap);
-
-  // draw some emblems for the item if needed
-  // we only support symlink emblem at the moment
-  Fm::FileInfo file = static_cast<FmFileInfo*>(index.data(Fm::FolderModel::FileInfoRole).value<void*>());
-  if(!file.isNull()) {
-    if(file.isSymlink()) {
-      painter->drawPixmap(iconPos, symlinkIcon_.pixmap(opt.decorationSize / 2, iconMode));
+    QPoint iconPos(opt.rect.x() + (opt.rect.width() - opt.decorationSize.width()) / 2, opt.rect.y());
+    QPixmap pixmap = opt.icon.pixmap(opt.decorationSize, iconMode);
+    painter->drawPixmap(iconPos, pixmap);
+
```

Comment:
All instances of `opt.decorationSize` should be changed to `option.decorationSize` and the above three lines should be changed as in `folderitemdelegate.cpp` because otherwise, the alignment of small icons/thumbnails and their text rectangles would be incorrect on desktop.

I don't why this issue didn't show up before your PR but if you want, I can fix it with a commit.